common/cache/namespaceCache.go

```diff
@@ -492,16 +501,34 @@ UpdateLoop:
 	c.cacheByID.Store(newCacheByID)
 	c.cacheNameToID.Store(newCacheNameToID)
 	c.triggerNamespaceChangeCallbackLocked(prevEntries, nextEntries)
+
+	// only update last refresh time when refresh succeeded
+	c.lastRefreshTime.Store(now)
 	return nil
 }
 
-func (c *namespaceCache) checkNamespaceExists(
+func (c *namespaceCache) checkAndRefresh(
 	name string,
 	id string,
-) error {
+) (bool, error) {
+	now := c.timeSource.Now()
+	if now.Sub(c.lastRefreshTime.Load().(time.Time)) < NamespaceCacheMinRefreshInterval {
+		return false, nil
+	}
 
+	c.checkLock.Lock()
+	defer c.checkLock.Unlock()
+
+	now = c.timeSource.Now()
```

Comment:
Do you really need it one more time here?

Reply:
yes, consider 1000 caller with 10 frontend host with namespace not registered.

on average, there will be ~ 100 concurrent call to this cache, so the line 514 line `now := c.timeSource.Now()` maybe called a while back.